File: app/src/main/java/com/example/shaochengyang/zillow/data/DataManager.java
package com.example.shaochengyang.zillow.data;

import android.content.Context;

import com.example.shaochengyang.zillow.data.network.INetworkHelper;
import com.example.shaochengyang.zillow.data.network.NetworkHelper;

public class DataManager implements IDataManager{

    INetworkHelper iNetworkHelper;

    public DataManager(Context context) {
        this.iNetworkHelper = new NetworkHelper();
    }

    @Override
    public void getPropertyList(onPropertyListListerner onPropertyListListerner, int i, String landlord) {
        iNetworkHelper.getPropertyList(onPropertyListListerner, 3, "landlord");
    }
}

File: app/src/main/java/com/example/shaochengyang/zillow/data/DataManager.kt
package com.example.shaochengyang.zillow.data

import android.content.Context

import com.example.shaochengyang.zillow.data.model.PropertyItem
import com.example.shaochengyang.zillow.data.network.NetworkHelper
import com.example.shaochengyang.zillow.viewmodel.ViewModel
class DataManager :IDataManager {


    companion object {
        val NetworkHelper : NetworkHelper = NetworkHelper()
    }
    override fun getPropertyList(i: Int, s: String, viewModel: ViewModel) {
        NetworkHelper.getPropertyList(i,s, viewModel)
    }

    override fun addProperty(propertyItem: PropertyItem, viewModel: ViewModel) {
        NetworkHelper.addProperty(propertyItem, viewModel)
    }

    override fun removeProperty(propertyid: String, viewModel: ViewModel) {
        NetworkHelper.removeProperty(propertyid, viewModel)
    }

    override fun getAllPropertyInfo(listener : IDataManager.onAllPropertyInfoListener) {
        NetworkHelper.getAllPropertyInfo(listener)
    }



}